src/library/Uno.Material/Controls/Chips/ChipGroup.cs

```diff
@@ -0,0 +1,332 @@
+	{
+		public event ChipItemEventHandler ItemClick;
+		public event ChipItemEventHandler ItemChecked;
+		public event ChipItemEventHandler ItemUnchecked;
+
+		private bool _isSynchronizingSelection = false;
+		private bool _isUpdatingSelection = false;
+
+		public ChipGroup()
+		{
+			RegisterPropertyChangedCallback(ItemsSourceProperty, (s, e) => (s as ChipGroup)?.OnItemsSourceChanged());
+
+			this.Loaded += OnLoaded;
+		}
+
+		protected override void OnApplyTemplate()
+		{
+			base.OnApplyTemplate();
+			ApplyThumbnailTemplate();
+		}
+
+		private void OnLoaded(object sender, RoutedEventArgs e)
+		{
+			EnforceSelectionMode();
+		}
+
+		private void OnSelectionMemberPathChanged(DependencyPropertyChangedEventArgs e)
+		{
+			var binding = SelectionMemberPath != null
+				? new Binding { Path = new PropertyPath(SelectionMemberPath), Mode = BindingMode.TwoWay }
+				: null;
+			foreach (var container in GetItemContainers())
+			{
+				if (binding != null)
+				{
+					container.SetBinding(Chip.IsCheckedProperty, binding);
+				}
+
+				container.ClearValue(Chip.IsCheckedProperty);
+			}
+		}
+
+		private void ApplyThumbnailTemplate()
+		{
+			if (ThumbnailTemplate != null)
+			{
+				foreach (var container in GetItemContainers())
+				{
+					container.Thumbnail = container.Content;
+					container.ThumbnailTemplate = ThumbnailTemplate;
+				}
+			}
+		}
+
+		private void ApplyCanRemoveProperty()
+		{
+			foreach (var container in GetItemContainers())
+			{
+				container.CanRemove = CanRemove;
+			}
+		}
+
+		private void OnSelectionModeChanged(DependencyPropertyChangedEventArgs e)
+		{
+			EnforceSelectionMode();
+		}
+
+		private void OnItemIsCheckedChanged(object sender, RoutedEventArgs e)
+		{
+			if (_isSynchronizingSelection) return;
+			if (sender is Chip container)
+			{
+				UpdateSelection(new[] { container });
+			}
+		}
+
+		protected override void OnItemsChanged(object e)
+		{
+			base.OnItemsChanged(e);
+
+			EnforceSelectionMode();
+		}
+
+		protected void OnItemsSourceChanged()
+		{
+			EnforceSelectionMode();
+		}
+
+		private void OnSelectedItemChanged(DependencyPropertyChangedEventArgs e)
+		{
+			if (_isSynchronizingSelection || _isUpdatingSelection) return;
+
+			if (SelectionMode == ChipSelectionMode.Single && FindContainer(SelectedItem) is Chip container)
+			{
+				container.SetIsCheckedSilently(true);
+				UpdateSelection(new[] { container });
+			}
+			else
+			{
+				UpdateSelection(null);
+			}
+		}
+
+		private void OnSelectedItemsChanged(DependencyPropertyChangedEventArgs e)
+		{
+			if (_isSynchronizingSelection || _isUpdatingSelection) return;
+
+			if (SelectionMode == ChipSelectionMode.Multiple)
+			{
+				var selectedContainers = SelectedItems != null
+					? GetItemContainers().Where(x => SelectedItems.Contains(x)).ToArray()
+					: null;
+				foreach (var container in selectedContainers ?? Enumerable.Empty<Chip>())
+				{
+					container.SetIsCheckedSilently(true);
```

Comment:
Why do do iterate in an Empty Enumerable?

Reply:
It will not iterate through it. I'm pretty sure it's a fallback value in case, there are no selected items.

Reply:
@Xiaoy312 can you confirm

Reply:
yes ?? is if the previous variable is null. If we `foreach(var container in selectedContainers)` and selectedContainers is null, it will throw an exception.
`foreach(var container in selectedContainers ?? Enumerable.Empty<Chip>())` will not throw an exception. If selectedContainers is null, it will iterate an empty enumerable instead. (which will do nothing).

Reply:
@MaximeDion-Work  https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/operators/null-coalescing-operator

Reply:
Cant foreach on null, since it will NRE. You either dont run it by wrapping a `if`-check or coalesce with empty. The latter has less nesting and indentation, which is why it is favored.